### PR TITLE
Remove more "ADClient" words

### PR DESCRIPTION
**Pull request by taquitos**
Rename `ad` to `affiche`
Apple might have gotten more paranoid about Ad tracking classes being available in children's apps.

### CODE REVIEW

File: Purchases/Logging/Strings/AttributionStrings.swift [hunk 1]
```diff
@@ -84,7 +84,7 @@ extension AttributionStrings: CustomStringConvertible {
         case .att_framework_present_but_couldnt_call_tracking_authorization_status:
             return "ATT Framework was found but it didn't respond to authorization status selector!"
 
-        case .iad_framework_present_but_couldnt_call_request_attribution_details:
+        case .apple_affiche_framework_present_but_couldnt_call_request_attribution_details:
             return "iAd Framework was found but it didn't respond to attribution details request!"
```
Comment (vegaro) [inline Purchases/Logging/Strings/AttributionStrings.swift:88]: what about this string?

Comment (taquitos) [reply Purchases/Logging/Strings/AttributionStrings.swift:88]: Ooops

Comment (taquitos) [reply Purchases/Logging/Strings/AttributionStrings.swift]: Done ✅

File: Purchases/Attribution/AfficheClientProxy.swift [hunk 2]
```diff
@@ -20,39 +20,39 @@ typealias AttributionDetailsBlock = ([String: NSObject]?, Error?) -> Void
 // Review uses some grep to find the class names, so we ended up creating a fake class that
 // exposes the same methods we're looking for in ADClient to call the same methods and mangling
 // the class names. So that Apple can't find them during the review, but we can still access them on runtime.
-class FakeAdClient: NSObject {
+class FakeAfficheClient: NSObject {
```
Comment (aboedo) [inline Purchases/Attribution/AfficheClientProxy.swift:23]: putting that thesaurus to use 😂 
I feel like future devs might have a hard time figuring out which class we're faking. 
Now, of course, we could add a link to the original class, which would be https://developer.apple.com/documentation/iad/adclient?language=objc, but that has the word `adclient` in it 🤦 

So how about we add a link, but do it through our shortener, i.e.: `rev.cat/fake-affice-client` and have that redirect to https://developer.apple.com/documentation/iad/adclient?language=objc? 

That way we could add a comment like: 
```swift
// This class is a fake of https://rev.cat/fake-affice-client. We need this class ...
```
same thought for the other stuff

**Review by aboedo — APPROVED** [review]
looks great! although I think we should add a comment so it's easy to map these names back to their original classes, I left a suggestion for that

**Review by vegaro — APPROVED** [review]

File: Purchases/Attribution/AfficheClientProxy.swift (hunk 2 above)
Comment (taquitos) [reply Purchases/Attribution/AfficheClientProxy.swift:23]: I love this idea.

Comment (taquitos) [reply Purchases/Attribution/AfficheClientProxy.swift]: Done!

**Pull request merged**